src/lib/types.ts

```diff
@@ -0,0 +1,31 @@
+export interface FetchData {
+  page: number;
+  results: Array<{
+    id: number;
+    backdrop_path: string;
+    poster_path: string;
+    title: string;
+    overview: string;
+    adult: boolean;
+    original_language: string;
+    original_title: string;
+    popularity: number;
+    release_date: string;
+    video: boolean;
+    vote_average: number;
+    vote_count: number;
+  }>;
+}
+
+export interface PreviewrProps {
+  fetchData: FetchData;
+}
```

Comment:
보니까 PreviewProps 에서는 id, title, poster_path 이정도만 사용되는 것 같더라고요??

FetchData 인터페이스를 모두 부여할 필요 없이 유틸리티 타입을 활용해서 유연하게 대처할 수 있을 것 같아요!!
(https://kyounghwan01.github.io/blog/TS/fundamentals/utility-types/)

```suggestion
 interface MovieInfo{
    id: number;
    backdrop_path: string;
    poster_path: string;
    title: string;
    overview: string;
    adult: boolean;
    original_language: string;
    original_title: string;
    popularity: number;
    release_date: string;
    video: boolean;
    vote_average: number;
    vote_count: number;
   }
   
   export interface FetchData{
     page:number;
     results: MovieInfo[];
   }
   
   export interface PreviewrProps{
      fetchData:  Pick<MovieInfo, 'id' | 'title' | 'poster_path'>[]
   }
```